pkg/cri/server/sandbox_run.go

```diff
@@ -218,16 +192,28 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		containerd.WithContainerExtension(sandboxMetadataExtension, &sandbox.Metadata),
 		containerd.WithRuntime(ociRuntime.Type, runtimeOpts)}
 
```

Comment:
We should update the state machine in https://github.com/containerd/containerd/blob/902212651b12e4d81b7bc779632b2e4da0a8e673/pkg/cri/store/sandbox/status.go

Reply:
agree .. and that may be a problem we are already seeing context timeouts on retrieval of status between creating the pod container and starting it.. as the start may take a while on a system that is resource constrained (yes even the pause container :-)

Reply:
Good callout. Let me update that graph.